velox/docs/functions/spark/aggregate.rst

```diff
@@ -58,6 +58,31 @@ General Aggregate Functions
     Returns an array created from the input ``x`` elements. Ignores null
     inputs, and returns an empty array when all inputs are null.
 
+.. spark:function:: collect_set(x) -> array<[same as x]>
+
+    Returns an array consisting of all unique values from the input ``x`` elements. 
+    Null values are excluded, and returns an empty array when all inputs are null.
+    Nested nulls are compared as values.
+    Map type input is not supported as it's not orderable type in Spark.
```

Comment:
I wonder why maps are not supported. I understand that maps are not orderable, but there is no ordering happening in this function.

Reply:
Sorry, map type should be not comparable in Spark.

Reply:
Spark uses compare to conduct ordering, are they the same in Spark? Thanks.

Reply:
This function is implemented with `HashSet` in Spark, which requires key should be comparable.
This should be my typo, I mean "Map type input is not supported as it's not comparable type in Spark." and updated it.
I added test cases to make sure this function has same behavior with Spark, specifically for some special cases like Nan, null, nested null.

Reply:
Got it. Thanks.